Added LSP-serenata

mdeboer opened the pull request:
PHP support for Sublime's LSP plugin using [Serenata](https://serenata.gitlab.io).

mdeboer commented:
> ### Automated testing result: WARNING
> 
> Repo link: [LSP-serenata](https://github.com/Cloudstek/LSP-serenata)
> [Results help](https://github.com/packagecontrol/st_package_reviewer/wiki/Package-checks)
> 
> ```
> Packages added:
>   - LSP-serenata
> 
> Processing package "LSP-serenata"
>   - WARNING: '.no-sublime-package' is defined. Please verify that it is *really* necessary
> ```

It is, as the server binary (phar file) is downloaded at runtime :)

rwols commented:
Perhaps you can use `os.path.join(sublime.cache_path(), "LSP-serenata")` to put the server binary there?

mdeboer commented:
Ah good idea, let me try that thanks!

mdeboer commented:
@rwols I have made the changes in 0.0.2 👍 

@packagecontrol-bot

FichteFoll commented:
Please resolve the merge conflict. Otherwise this is ready to merge.

FichteFoll commented:
Actually, it's not fine, because `package_path` is function-local when it is being set [here](https://github.com/Cloudstek/LSP-serenata/blob/65400c39b5576cef45196c98b6c6c51f64186cf0/plugin.py#L45). You must use the `global` keyword first.

mdeboer commented:
Please check again, I have made changes as the cache path even when set globally is not always available apparently. The package works fine.

rwols commented:
The cache path should always exist. It's just that the behavior of the `sublime` python module is undefined when `plugin_loaded` is not yet invoked. Calling functions from the `sublime` module after `plugin_loaded` should be consistent.

FichteFoll commented:
You haven't made any changes [since my last comment](https://github.com/Cloudstek/LSP-serenata/commits/master). There's still a usage of `package_path` without proper initialization [here](https://github.com/Cloudstek/LSP-serenata/blob/65400c39b5576cef45196c98b6c6c51f64186cf0/plugin.py#L64) (and `server_path` below that).

(Seriously, just use the `global` keyword within `plugin_loaded`.)

Also, please resolve the merge conflict.

mdeboer commented:
I will have a go at it @rwols, thanks for clearing that up. And yes, I will resolve that merge conflict @FichteFoll, calm down.

mdeboer commented:
As LSP requires the cache path before `plugin_loaded` has been called, a global could not be used. However I implemented a fix which works and cleaned it all up. Would appreciate if it can be merged now.

FichteFoll commented:
I don't know about LSP internals, but LGTM.
